app/controllers/sessions_controller.rb

```diff
@@ -20,7 +20,7 @@ def callback
     if current_user
       DfESignInUsers::Update.call(user: current_user, dfe_sign_in_user: dfe_sign_in_user)
 
-      redirect_to root_path
+      redirect_to(session[:redirect_back_to] ? session.delete(:redirect_back_to) : root_path)
```

Comment:
```suggestion
      redirect_to session.delete(:redirect_back_to) || root_path
```
`Hash#delete` returns the value or `nil` if it isn't present.